### ISSUE TITLE
Feature Request: Add option to use read replicas by default for count operations

### ISSUE THREAD

**Issue by ashishra0** [magnusvk/counter_culture#403]
## Problem
Currently, when using counter_culture to fix counts, all read operations (including counting records) happen on the primary database. This can cause significant lock contentions and performance issues, especially when running counter_culture_fix_counts in background jobs with large datasets.

## Current Workaround
We currently have to explicitly specify using read replicas for each counter_culture_fix_counts call:

```ruby
Model.counter_culture_fix_counts(
  db_connection_builder: proc { |reading, block|
    if reading
      ApplicationRecord.connected_to(role: :reading, &block)
    else
      ApplicationRecord.connected_to(role: :writing, &block)
    end
  }
)
```

I was thinking something along these lines

```ruby
# In initializer
CounterCulture.configure do |config|
  config.use_read_replica = true
end

# Or as a default option in counter_culture itself
```

Apologies for asking if this feature already exists

**Comment by magnusvk** [comment]
Yeah, that does seem like a simpler option, and I don't think it exists yet. 

I won't be able to work on a change like that myself, but would be happy to look at a PR implementing this.

**Comment by ashishra0** [comment]
Okay great, I will try and spin up a PR for this! Thanks :)
